### PR TITLE
Fix/generate token

### PR DESCRIPTION
**Pull request by vigan-abd**
### Description:
This PR adds missing `caps` param to `generateToken` method

### Breaking changes:
- [ ]

### New features:
- [ ]

### Fixes:
- fix: missing caps param in generateToken

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [x] Tests added or updated
- [x] Documentation updated

### CODE REVIEW

File: lib/rest2.js [hunk 1]
```diff
@@ -1221,15 +1221,17 @@ class RESTv2 {
...
+   * @param {boolean} [opts.writePermission] - token write permission
+   * @param {string} [opts._cust_ip] - user ip address
    * @param {Function} [cb] - callback
    * @returns {Promise} p
    */
-  generateToken ({ ttl, scope, writePermission, _cust_ip: custIp } = {}, cb) {
+  generateToken ({ ttl, scope, caps, writePermission, _cust_ip: custIp } = {}, cb) {
     return this._makeAuthRequest('/auth/w/token', {
-      ttl, scope, writePermission, _cust_ip: custIp
+      ttl, scope, caps, writePermission, _cust_ip: custIp
```
Comment (robertkowalski) [inline lib/rest2.js:1234]: hmm, not so nice:

```
{
  ttl: undefined,
  scope: 'blerg',
  caps: undefined,
  writePermission: undefined,
  _cust_ip: undefined
}
```

for 

```js
  const response = await rest2.generateToken({
    scope: 'blerg' // only required param
  })
```

File: lib/rest2.js [hunk 2]
```diff
@@ -1229,10 +1231,12 @@ class RESTv2 {
    * @param {Function} [cb] - callback
    * @returns {Promise} p
    */
-  generateToken ({ ttl, scope, caps, writePermission, _cust_ip: custIp } = {}, cb) {
-    return this._makeAuthRequest('/auth/w/token', {
-      ttl, scope, caps, writePermission, _cust_ip: custIp
-    }, cb)
+  generateToken (opts, cb) {
+    let params = _pick(opts || {}, ['ttl', 'scope', 'caps', 'writePermission', '_cust_ip'])
+    params = _pickBy(params, Boolean) // clear falsy values
```
Comment (robertkowalski) [inline lib/rest2.js:1236]: this does not work with writePermission, which is a boolean:

```js
  const response = await rest2.generateToken({
    scope: 'blerg',
    writePermission: false
  })
```

Comment (vigan-abd) [reply lib/rest2.js]: adjusted the pickBy filter to check nil values

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**